posts/integrating-civic-into-a-static-serverless-website-part-2-of-2.md

```diff
@@ -0,0 +1,443 @@
+
+`npm install -g serverless`
+
+p.s. You will be able to use `sls` and `serverless` commands interchangeably.
+
+We could use a template to generate the directory and config file, which would take the command `sls create --template aws-nodejs --path ServerlessCivicApp`, but we're gonna change almost everything generated. You could see Serverless's [list of templates](https://github.com/serverless/serverless/tree/master/lib/plugins/create/templates) for possible template options. They support a large range of languages and hosting services.
+
+Our project structure should look like the following.
+```text
+ServerlessCivicApp: 
+  - functions
+      - auth.js
+      - color.js
+  - package.json
+  - serverless.yml
+```
+
+`package.json` just needs to keep track of our dependencies. (don't forget to `npm install`!)
+
+```json
+{
+  "name": "serverless-civic-app",
+  "version": "1.0.0",
+  "dependencies": {
+    "civic-sip-api": "^1.1.0",
+    "jsonwebtoken": "^8.4.0"
+  },
+  "devDependencies": {
+    "serverless-domain-manager": "^2.6.13"
+  }
+}
+```
+
+`serverless.yml` lets us configure the functions, environments, AWS settings, etc. [This Serverless article](https://serverless.com/framework/docs/providers/aws/events/apigateway/) is probably the best resource for understanding the layout of this file. 
+
+```yaml
+service: my-serverless-civic-app
+
+provider:
+  name: aws
+  runtime: nodejs8.10
+  stage: dev
+  region: us-east-1
+
+functions:
+  color-identification:
+    handler: functions/color.colorIdentification
+  custom-civic-authorizer:
+    handler: functions/auth.customCivicAuthorizer
+```
+
+Here is the core of the config file. We will be adding more later too.
+- `service` is just the title that is used to generate various resource names in AWS. It should be descriptive and will tie each deployment together (if you keep deploying from the same project).
+- `provider` lets Serverless know where to deploy our resources. `name` and `runtime` are the only ones required, as the others have defaults. 
+- `functions` lists out our lambdas. We only have two so you can see where Serverless picks up the function handlers. Those functions will need to be exported from each file as we will see later.
+
+### Write our Lambda authorizer
+
+A good resource that AWS provides for learning how to define Lambda functions using NodeJS can be found [here](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-handler.html). The main pieces of information that we need to know are what the function signatures should be and how to return or error out of the function. All of the code here will be taking advantage of [ES2017 async and await](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function), but there is no problem with refactoring with promises and callbacks.
```

Comment:
```suggestion
A good resource that AWS provides for learning how to define Lambda functions using NodeJS can be found [here](https://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-handler.html). The main pieces of information that we need to know are what the function signatures should be and how to return or error out of the function. All of the code here will be taking advantage of [ES2017 async and await](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function), but this can be refactored to use promises and callbacks instead.
```